lib/chef/provider/group/suse.rb

```diff
@@ -51,7 +52,12 @@ def set_members(members)
         end
 
         def add_member(member)
-          shell_out!("groupmod -A #{member} #{@new_resource.group_name}")
+          begin
+            Etc.getpwnam(member)
+            shell_out!("groupmod -A #{member} #{@new_resource.group_name}")
+          rescue ArgumentError => e
+            raise Chef::Exceptions::Group, "Could not add #{member} to #{@new_resource.group_name}: user does not exist"
```

Comment:
For the why-run case, I think that's what you want, e.g., Chef should say that the user doesn't exist, assuming it would have been created by a prior step.